### PR TITLE
Switch MongoDB tests to use Docker

### LINKED ISSUE #2229: Test MongoDB appender against Docker image
Right now (after  #1589), we are using Embedded MongoDB to:

- download a MongoDB distribution on our system,
- run it with the correct parameters,
- test the MongoDB appender.

This procedure has some problems:

1. we cannot run tests in parallel on a clean system (each test tries to download MongoDB at the same time),
2. the MongoDB distribution depends on some system libraries that are not always present.

Therefore we should move the tests to use [TestContainers](https://testcontainers.com/) or an equivalent solution and the official [Docker `mongo`](https://hub.docker.com/_/mongo/) images.

### PR DESCRIPTION
**Pull request by ppkarwasz**
MongoDB is a binary server. The current `log4j-mongodb` tests download a **generic** binary MongoDB distribution and try to run it. The distribution is not self-contained and requires several libraries (e.g., OpenSSL) to be available on the target host.

Those libraries are not always available in the required version: e.g., currently MongoDB needs OpenSSL 1, but OpenSSL 3 is bundled in the most recent Debian.

This PR switches from the binary distribution to the usage of the **latest** Docker image available.

The advantages of this approach are:

- We always test against the newest server version available.
- The success of the tests does not depend on the libraries installed on the host.
- Tests can run in parallel. In the current approach, parallel tests failed since each one tried to download MongoDB separately.

The main disadvantage is that Docker will be required to test `log4j-mongodb`. This is the case for the CI, but individual developers might need to install it too.

Fixes #2229.

### CODE REVIEW

**Review by garydgregory — CHANGES_REQUESTED** [review]
Nice improvement @ppkarwasz 😀

**Comment by ppkarwasz** [comment]
> Nice improvement @ppkarwasz 😀

Does it mean you approve? If not, what changes are you expecting?

**Review by garydgregory — APPROVED** [review]
Fat fingers! I approve.

File: log4j-mongodb/src/test/java/org/apache/logging/log4j/mongodb/MongoDbAuthFailureIT.java
Comment (vy) [inline log4j-mongodb/src/test/java/org/apache/logging/log4j/mongodb/MongoDbAuthFailureIT.java:33]: Can we suppress the output of this test, please? Tried adding `@UsingStatusListener`, but did not help.

**Review by vy — CHANGES_REQUESTED** [review]
Locally verified the PR. Dropped some minor requests/questions.

**Question:** Will this get ported to `2.x`?

I will align the `<activation>` of the `docker` profile of `log4j-template-layout-json-test` with this PR, once this gets merged.

File: log4j-mongodb/pom.xml
Comment (ppkarwasz) [reply log4j-mongodb/pom.xml:125]: Previously Surefire didn't run in parallel to prevent the download of MongoDB multiple times.
Now we don't have any unit tests any more, so we can skip Surefire.

File: log4j-mongodb/pom.xml
Comment (ppkarwasz) [reply log4j-mongodb/pom.xml:160]: What activation do you propose? On the presence of `/var/run/docker.sock`?

File: log4j-mongodb/pom.xml
Comment (ppkarwasz) [reply log4j-mongodb/pom.xml:262]: The `docker-maven-plugin` creates it. I added a comment in https://github.com/apache/logging-log4j2/pull/2922/commits/bbd4e7df620331c297049035cd70488ce17d5aeb

File: log4j-mongodb/pom.xml
Comment (ppkarwasz) [reply log4j-mongodb/pom.xml]: I replaced the `CI`-check with a check for a `/var/run/docker.sock` file: https://github.com/apache/logging-log4j2/pull/2922/commits/bbd4e7df620331c297049035cd70488ce17d5aeb

File: log4j-mongodb/src/test/java/org/apache/logging/log4j/mongodb/MongoDbAuthFailureIT.java
Comment (ppkarwasz) [reply log4j-mongodb/src/test/java/org/apache/logging/log4j/mongodb/MongoDbAuthFailureIT.java:36]: I silenced the tests in https://github.com/apache/logging-log4j2/pull/2922/commits/bbd4e7df620331c297049035cd70488ce17d5aeb.

`@UsingStatusListener` does not silence the status logger, it keeps its level the same as configured. In case of failure, however, it prints all the messages at least as severe `DEBUG`

Fixed in https://github.com/apache/logging-log4j2/pull/2922/commits/bbd4e7df620331c297049035cd70488ce17d5aeb

File: log4j-mongodb/pom.xml
Comment (vy) [reply log4j-mongodb/pom.xml:183]: @ppkarwasz, I did not have an objection to the previous activation condition, as a matter of fact, I support it. I just wanted to double-check if this was a conscious decision, since it won't run by default on the machine of the person-who-I-dont-want-to-call-his-name. :sweat: Reverted back to the original activation in  c8f156c67e2cd4ec28ee8b7992a1515418bc38c0.

**Review by vy — APPROVED** [review]

**Review by vy — APPROVED** [review]

**Comment by ppkarwasz** [comment]
Oh, tests are stuck, it must be Friday 13th again! :wink:

**Pull request merged**
